From 7a3b83705d3dae5572146e08b50cd6bf83557166 Mon Sep 17 00:00:00 2001
From: Carlos Valiente <carlos.valiente@ecmwf.int>
Date: Mon, 5 Jul 2010 09:21:09 +0100
Subject: [PATCH] Test PTYProcess

Previoues versions of the test were not passing the ``usePTY=True``
argument to ``reactor.spawnProcess``.
---
 twisted/internet/test/test_process.py |   31 ++++++++++++++++++-------------
 1 files changed, 18 insertions(+), 13 deletions(-)

diff --git a/twisted/internet/test/test_process.py b/twisted/internet/test/test_process.py
index 6eaf537..b356359 100644
--- a/twisted/internet/test/test_process.py
+++ b/twisted/internet/test/test_process.py
@@ -238,24 +238,29 @@ class ProcessTestsBuilderBase(ReactorBuilder):
             "sys.stdout.write(str(process._listOpenFDs()))",
             "sys.stdout.flush()")
 
+        def checkOutput(output):
+            self.assertEquals('[0, 1, 2, 3]', output)
+
         reactor = self.buildReactor()
 
-        def processFinished(output):
-            self.assertEquals('[0, 1, 2, 3]', output)
+        class Protocol(ProcessProtocol):
 
-        def shutdown(result):
-            reactor.stop()
-            return result
+            def __init__(self):
+                self.output = []
 
-        def spawnChild():
-            msg("Spawning child with Twisted at %r" % (top.path,))
-            d = succeed(None)
-            d.addCallback(lambda dummy: utils.getProcessOutput(
-                sys.executable, ["-c", "\n".join(source)], reactor=reactor))
-            d.addCallback(processFinished)
-            d.addBoth(shutdown)
+            def outReceived(self, data):
+                self.output.append(data)
 
-        reactor.callWhenRunning(spawnChild)
+            def processEnded(self, reason):
+                try:
+                    checkOutput("".join(self.output))
+                finally:
+                    reactor.stop()
+
+        proto = Protocol()
+        reactor.callWhenRunning(
+            reactor.spawnProcess, proto, sys.executable,
+            [sys.executable, "-c", "\n".join(source)], usePTY=self.usePTY)
         self.runReactor(reactor)
 
 
-- 
1.7.1

